src/AppInstallerCommonCore/Versions.cpp

```diff
@@ -275,4 +355,77 @@ namespace AppInstaller::Utility
 
         return m_version < other.m_version;
     }
+
+    VersionRange::VersionRange(Version minVersion, Version maxVersion)
+    {
+        THROW_HR_IF(E_INVALIDARG, minVersion > maxVersion);
+        m_minVersion = std::move(minVersion);
+        m_maxVersion = std::move(maxVersion);
+    }
+
+    bool VersionRange::HasOverlapWith(const VersionRange& other) const

```

Comment:
```suggestion
    bool VersionRange::Overlaps(const VersionRange& other) const
```
Why use many word, when few word do trick?